src/internals/ir/callGraph.ts

```diff
@@ -0,0 +1,169 @@
+    return this.nodeMap;
+  }
+
+  public getEdges(): Map<CGEdgeId, CGEdge> {
+    return this.edgesMap;
+  }
+
+  build(astStore: TactASTStore): CallGraph {
+    this.addFunctionsToNodes(astStore);
+    this.analyzeFunctionCalls(astStore);
+    return this;
+  }
+
+  private addFunctionsToNodes(astStore: TactASTStore) {
+    for (const func of astStore.getFunctions()) {
+      const funcName = this.getFunctionName(func);
+      if (funcName) {
+        const node = new CGNode(func.id, funcName);
+        this.nodeMap.set(node.idx, node);
+        this.nameToNodeId.set(funcName, node.idx);
+      }
+    }
+  }
+
+  private analyzeFunctionCalls(astStore: TactASTStore) {
+    for (const func of astStore.getFunctions()) {
+      const callerId = this.getNodeIdByName(this.getFunctionName(func));
+      if (callerId !== undefined) {
+        this.processFunctionBody(func, callerId);
+      }
+    }
+  }
+
+  private getFunctionName(
+    func: AstFunctionDef | AstReceiver | AstContractInit,
+  ): string {
+    switch (func.kind) {
+      case "function_def":
+        return func.name.text;
+      case "contract_init":
+        return `contract_init_${func.id}`;
+      case "receiver":
+        return `receiver_${func.id}`;
+      default:
+        unreachable(func);
+    }
+  }
+
+  private getNodeIdByName(name: string): CGNodeId | undefined {
+    return this.nameToNodeId.get(name);
+  }
+
+  private processFunctionBody(
+    func: AstFunctionDef | AstReceiver | AstContractInit,
+    callerId: CGNodeId,
+  ) {
+    forEachExpression(func, (expr) => {
+      this.processExpression(expr, callerId);
+    });
+  }
+
+  private processExpression(expr: AstExpression, callerId: CGNodeId) {
+    if (expr.kind === "static_call") {
+      const staticCall = expr as AstStaticCall;
+      const functionName = staticCall.function.text;
+      const calleeId = this.findOrAddFunction(functionName);
+      this.addEdge(callerId, calleeId);
+    } else if (expr.kind === "method_call") {
+      const methodCall = expr as AstMethodCall;
+      const methodName = methodCall.method.text;
+      const calleeId = this.findOrAddFunction(methodName);
+      this.addEdge(callerId, calleeId);
+    }
+  }
+
+  private findOrAddFunction(name: string): CGNodeId {
+    const nodeId = this.nameToNodeId.get(name);
+    if (nodeId !== undefined) {
+      return nodeId;
+    }
+
+    const newNode = new CGNode(undefined, name);
+    this.nodeMap.set(newNode.idx, newNode);
+    this.nameToNodeId.set(name, newNode.idx);
+    return newNode.idx;
+  }
+
+  private addEdge(src: CGNodeId, dst: CGNodeId) {
+    const srcNode = this.nodeMap.get(src);
+    const dstNode = this.nodeMap.get(dst);
+    if (srcNode && dstNode) {
+      const edge = new CGEdge(src, dst);
+      this.edgesMap.set(edge.idx, edge);
+      srcNode.outEdges.add(edge.idx);
+      dstNode.inEdges.add(edge.idx);
+    }
+  }
+
+  areConnected(src: CGNodeId, dst: CGNodeId): boolean {
```

Comment:
We need docstrings here because these are public functions used directly by the detector's developers.

Reply:
Also:
```suggestion
  public areConnected(src: CGNodeId, dst: CGNodeId): boolean {
```